src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/TargetFrameworkMoniker.cs

```diff
@@ -2,25 +2,200 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Text;
 
 namespace Microsoft.DotNet.UpgradeAssistant
 {
-    public record TargetFrameworkMoniker(string Name)
+    public record TargetFrameworkMoniker
     {
-        public static TargetFrameworkMoniker NetStandard20 { get; } = new TargetFrameworkMoniker("netstandard2.0");
+        private const string NetStandard = "netstandard";
+        private const string NetFramework = "netframework";
+        private const string Net = "net";
+        private const string NetCoreApp = "netcoreapp";
 
-        private const string NetStandardNamePrefix = "netstandard";
+#pragma warning disable CA1707 // Identifiers should not contain underscores
+        public static readonly TargetFrameworkMoniker NetStandard10 = new(NetStandard, new Version(1, 0));
+        public static readonly TargetFrameworkMoniker NetStandard20 = NetStandard10 with { FrameworkVersion = new Version(2, 0) };
+        public static readonly TargetFrameworkMoniker NetStandard21 = NetStandard20 with { FrameworkVersion = new Version(2, 1) };
+        public static readonly TargetFrameworkMoniker NetCoreApp21 = new(NetCoreApp, new Version(2, 1));
+        public static readonly TargetFrameworkMoniker NetCoreApp30 = NetCoreApp21 with { FrameworkVersion = new Version(3, 0) };
+        public static readonly TargetFrameworkMoniker NetCoreApp31 = NetCoreApp30 with { FrameworkVersion = new Version(3, 1) };
+        public static readonly TargetFrameworkMoniker Net45 = new(Net, new Version(4, 5));
+        public static readonly TargetFrameworkMoniker Net46 = Net45 with { FrameworkVersion = new Version(4, 6) };
+        public static readonly TargetFrameworkMoniker Net461 = Net45 with { FrameworkVersion = new Version(4, 6, 1) };
```

Comment:
nit: uses a different pattern than established above
```
NetStandard20 = Netstandard10
NetStandard21 = NetStandard20
```

vs

```
Net46 = Net45
Net461 = Net45
...
Net48 = Net45
```

Reply:
Cool. I've updated it.